### ISSUE TITLE
Select all button in Journal has no effect.

### ISSUE THREAD

**Issue by chimosky** [sugarlabs/sugar#945]
**Steps to reproduce**
* Search for existing files.
* Select one then use the `Select all` button above items.
* Selections are deselected instead of being selected.

**Comment by sourabhaa** [comment]
One more observation, this happens only if the name of the file(as mentioned inside activity.info)  is not entered completely in the search-box.
Example : To select ```terminal activity```, typing at least 'terminal' will give expected results. Anything less than that like 'ter' will result in this error.
Also, I noticed if complete name of an activity is searched, the auto search result will include all other activities.

**Comment by chimosky** [comment]
> One more observation, this happens only if the name of the file(as mentioned inside activity.info) is not entered completely in the search-box.

Thanks for that, just noticed this too and it shouldn't have to be the case. @quozl was there any particular reason for this?

>  Also, I noticed if complete name of an activity is searched, the auto search result will include all other activities.

That's expected given what you mentioned above.

**Comment by quozl** [comment]
I tried but can't reproduce.  Can you explain further what happens and what you expected?

**Comment by chimosky** [comment]
> I tried but can't reproduce. Can you explain further what happens and what you expected?

When I search for `TamTam` all the TamTam instances show in the search results and after selecting one, I use the select all button and I expect that everything that came up after the search result be selected but what happens is the selected activity is deselected and the search is automatically cleared.

**Comment by sourabhaa** [comment]
For more clarity,
1. Search  ```terminal``` , all search history of ```terminal activity``` is listed, ```Select all``` feature works fine (because inside activity.info, ``name = Terminal``). 
2. Search ```term```, all search history of ```terminal activity``` is listed, ```Select all``` button deselects.
3. Search ```terminal activity```, all search history of **all** the activities(which has 'activity' in its name as shown in the search list (not in activity.info)) is listed, ```Select all``` button works fine(selects all the activities listed).

> That's expected given what you mentioned above.

The third case is what I meant. You mean both 2nd and 3rd case have same cause?

> Hope you're doing okay @sourabhaa, been hearing about what's been happening in India.

*Thanks for asking @chimosky , I'm in perfect health (but the situation is worsening...)*

**Comment by chimosky** [comment]
> The third case is what I meant. You mean both 2nd and 3rd case have same cause?

No, I mean when you search for an activity and include the words "Activity" all activities are bound to show up in the result as they have that in their name when saved as journal objects.

> I'm in perfect health (but the situation is worsening...)

Glad you're okay, I hope things change.

**Comment by quozl** [comment]
I guess I'm still confused.  Are you searching activities or Journal objects?

**Comment by quozl** [comment]
Okay, searching for Journal objects that happen to contain activity
names because the user did not change the object name.

I don't remember the details of the search entry changes.  Best would
be to reverse engineer the code.  Sam Parkinson added completion.

**Comment by Vishalk91-4** [comment]
@quizol is this issue `open` still 
For me to work on
